ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs

```diff
@@ -140,39 +147,42 @@ scheduleCountSkips (Schedule (t:ts)) = go t ts
 --
 -- Returns the set of slot numbers that 'BlockchainTime' should report or,
 -- if time moved backwards, the @(before, after)@ slot pair where @after@ is
--- (strictly) less than @before@.
+-- more than the @maxClockRewind@ less than @before@.
 --
--- NOTE: Assumes the slot length is 1 for these sets.
+-- NOTE: Assumes the slot length is 1 and max clock rewind is 2 for these sets.
 model :: Int -> Schedule -> Either (SlotNo, SlotNo) [SlotNo]
-model = \need (Schedule s) -> runExcept $ go need s (SlotNo 0)
+model = \need ss ->
+    runExcept $ (SlotNo 0 :) <$> go (need - 1) ss (0.0, SlotNo 0)
   where
-    go :: Int        -- How many slots do we still need to collect?
-       -> [Fixed E1] -- Remaining schedule
-       -> SlotNo     -- Current slot
-       -> Except (SlotNo, SlotNo) [SlotNo]
-
-    -- No more slots required
-    go 0 _ _ =
-        return []
-
-    -- If we don't override the delays, everything just works as expected
-    go need [] now =
-        return [SlotNo (unSlotNo now + n) | n <- take need [0 ..]]
-
-    go need (s:ss) now
-      -- Time didn't actually move according to the schedule, 'BlockchainTime'
-      -- should wait until it does.
-      | now' == now = go need ss now
-
-      -- If time did move forward, 'BlockchainTime' should report the next slot
-      -- (which might not be the successor of the previous)
-      | now' >  now = (now :) <$> go (need - 1) ss now'
-
-      -- If time went backwards, we should see an exception
-      | otherwise   = throwError (now, now')
+    -- This let's us treat the schedule as an infinite stream of offsets
+    advanceSchedule :: Schedule -> (Fixed E1, Schedule)
+    advanceSchedule (Schedule ss) =
+        case ss of
+          []    -> (0.0, Schedule [0.0])
```

Comment:
Move this up and replace with an error case, add a comment about the local invariant.